python/packages/ai/teams/ai/prompts/prompt_manager.py

```diff
@@ -20,18 +20,19 @@
 
 SK_CONFIG_FILE_NAME = "config.json"
 SK_PROMPT_FILE_NAME = "skprompt.txt"
+StateT = TypeVar("StateT", bound=TurnState)
 
 
-class PromptManager:
+class PromptManager(Generic[StateT]):
```

Comment:
Can you share why we need to use generic here? I may need an example to better understand.

Reply:
Like in the comment about the planner above, if the prompt manager doesn't use the same turn state as the ai/app class it will give type errors. If we want to remove generics we have to do it for the entire project so all the components are compatible. I will look into it today when I get some time.

Reply:
I'm good to go with your current implementation. But I wrote some examples here: https://github.com/microsoft/teams-ai/blob/1c886a283043b6bd2c1cee1455f08dfce3662c1a/python/packages/ai/examples/openai_planner.py#L45-L61. Not sure how you feels about it. It indeed introduces some difference with other SDK.

Reply:
That makes sense to me, I'm going to create a task to handle that work separately but I will send it to you and start it today so that all our state uses that method.